core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java

```diff
@@ -155,8 +161,8 @@ public InnerHitBuilder(StreamInput in) throws IOException {
         explain = in.readBoolean();
         version = in.readBoolean();
         trackScores = in.readBoolean();
-        fieldNames = (List<String>) in.readGenericValue();
-        fieldDataFields = (List<String>) in.readGenericValue();
+        storedFieldNames = (List<String>) in.readGenericValue();
```

Comment:
While you are here can you move this constructor (the reading one) below all the other constructors? That way it can be *right* next to `writeTo`. It is nice to be able to fit reading and writing onto the same screen.